### PR TITLE
refactor: simplify CertificateExpiryMonitor to remove queue

### PR DESCRIPTION
**Pull request by MikeDombo**
**Issue #, if available:**

**Description of changes:**
Remove overly complicated Queue which wasn't adding any value.

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sushantAtAws — DISMISSED** [review]

**Review by naikvidish — DISMISSED** [review]

File: src/main/java/com/aws/greengrass/certificatemanager/certificate/CertificateExpiryMonitor.java [hunk 1]
```diff
@@ -52,39 +45,28 @@ public CertificateExpiryMonitor(ScheduledExecutorService ses, ConnectivityInfoPr
...
 
     private void watchForCertExpiryOnce() {
-        List<CertificateGenerator> triedCertificateGenerators = new ArrayList<>();
-
-        for (CertificateGenerator cg = monitoredCertificateGenerators.peek(); cg != null;
-             cg = monitoredCertificateGenerators.peek()) {
+        for (CertificateGenerator cg : monitoredCertificateGenerators) {
             if (!cg.shouldRegenerate()) {
-                break;
+                continue;
```
Comment (sushantAtAws) [inline src/main/java/com/aws/greengrass/certificatemanager/certificate/CertificateExpiryMonitor.java:62]: wow. good catch

File: src/test/java/com/aws/greengrass/certificatemanager/certificate/CertificateExpiryMonitorTest.java [hunk 2]
```diff
@@ -35,17 +37,23 @@
...
 
     @Mock
     private ConnectivityInfoProvider mockConnectivityInfoProvider;
+    private final ScheduledExecutorService ses = new ScheduledThreadPoolExecutor(1);
 
     @TempDir
     Path tmpPath;
 
+    @AfterEach
+    void afterEach() {
```
Comment (sushantAtAws) [inline src/test/java/com/aws/greengrass/certificatemanager/certificate/CertificateExpiryMonitorTest.java:53]: If its expected to be there for EACH test, it needs to be initialized again. Its can be confusing for future devs.

**Review by sushantAtAws — APPROVED** [review]

File: src/test/java/com/aws/greengrass/certificatemanager/certificate/CertificateExpiryMonitorTest.java (hunk 2 above)
Comment (MikeDombo) [reply src/test/java/com/aws/greengrass/certificatemanager/certificate/CertificateExpiryMonitorTest.java:53]: No it doesn't need to be reinitialized.

**Comment by MikeDombo** [comment]
@sushantAtAws 
I just made another update which just adds `GGExtension` to all test classes that were missing it. That's the only chage.

**Review by naikvidish — APPROVED** [review]

**Pull request merged**